packages/react-components/react-utilities/src/compose/resolveShorthand.ts

```diff
@@ -33,7 +33,13 @@ export const resolveShorthand: ResolveShorthandFunction = (value, options) => {
   if (typeof value === 'string' || typeof value === 'number' || Array.isArray(value) || isValidElement(value)) {
     resolvedShorthand.children = value;
   } else if (typeof value === 'object') {
-    resolvedShorthand = value;
+    /**
+     * TODO: Figure out correct typing solution:
+     * Type 'UnknownSlotProps | ReactNodeArray' is not assignable to type 'UnknownSlotProps'.
+     * Type 'readonly ReactNode[]' has no properties in common with type 'UnknownSlotProps'.
+     */
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    resolvedShorthand = value as any;
```

Comment:
Not sure what happened here but this can be addressed in a separate PR by someone who may be more familiar with the typings

Reply:
Fixed by replacing deprecated `ReactNodeArray` with `ReactNode[]` [be73ad6](https://github.com/microsoft/fluentui/pull/24356/commits/be73ad625a837b012fbda5134914d79cc730bc5a)